python/samples/getting_started_with_processes/step01_processes.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+from typing import Generic, TypeVar
+
+from semantic_kernel import Kernel
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.connectors.ai.open_ai import OpenAIChatCompletion
+from semantic_kernel.contents import ChatHistory
+from semantic_kernel.functions import kernel_function
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.processes.kernel_process.kernel_process_step_context import KernelProcessStepContext
+from semantic_kernel.processes.kernel_process.kernel_process_step_state import KernelProcessStepState
+from semantic_kernel.processes.local_runtime.local_event import KernelProcessEvent
+from semantic_kernel.processes.process_builder import ProcessBuilder
+from semantic_kernel.processes.process_function_target_builder import ProcessFunctionTargetBuilder
+
+
+class CommonEvents:
+    UserInputReceived = "UserInputReceived"
+    AssistantResponseGenerated = "AssistantResponseGenerated"
+
+
+class ChatBotEvents:
+    StartProcess = "startProcess"
+    IntroComplete = "introComplete"
+    AssistantResponseGenerated = "assistantResponseGenerated"
+    Exit = "exit"
+
+
+class UserInputState(KernelBaseModel):
+    user_inputs: list[str] = []
+    current_input_index: int = 0
+
+
+# Create a type variable for the generic state
+TState = TypeVar("TState", bound=KernelBaseModel)
+
+
+class KernelProcessStep(Generic[TState]):
+    def __init__(self):
+        self._state: TState | None = None
+
+    async def activate(self, state: "KernelProcessStepState[TState]"):
+        """Activates the step and sets the state."""
+        pass
+
+    def create_default_state(self) -> TState:
+        """Creates the default state object."""
+        return None
+
+    async def on_activate(self):
+        """To be overridden by subclasses if needed."""
+        pass
+
+
+class ScriptedUserInputStep(KernelProcessStep[UserInputState]):
```

Comment:
Need to clean this up, too -- move it to an appropriate spot, and implement the latest change that you introduced last week in C#, @esttenorio.